Library/Homebrew/language/perl.rb

```diff
@@ -11,10 +11,12 @@ module Shebang
       module_function
 
       def detected_perl_shebang(formula = self)
-        perl_path = if formula.uses_from_macos_elements&.include? "perl"
-          "/usr/bin/perl#{MacOS.preferred_perl_version}"
-        elsif formula.deps.map(&:name).include? "perl"
+        perl_path = if formula.deps.map(&:name).include? "perl"
           Formula["perl"].opt_bin/"perl"
+        elsif formula.uses_from_macos_elements
+                     .flat_map { |e| e.is_a?(Hash) ? e.keys : e }
+                     .include? "perl"
```

Comment:
Thoughts on this for the `formula_auditor` check too?

Reply:
@Bo98 Hah, started on this before I saw the comment. Pulled into a method.